src/Chapter3.hs

```diff
@@ -481,7 +503,28 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
-
+data City = City
+  { castle :: Maybe String
+  , wall :: Bool
+  , buildings :: Buildings
+  , house :: [House] 
+  }
+
+newtype House = House { number_of_people :: Int}
```

Comment:
To avoid building houses with more than 4 people you need to restrict usage of the Constructor and allow to create `House`s only with your own `mkHouse` function.
You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Generally, in here, you could restrict that on type level, by using something else than `Int` 🙂